src/HybridSolver.cpp

```diff
@@ -463,22 +510,28 @@ void HybridSolver::RecoverOriginalSolution(const mfem::Vector& HybridSol,
 
 void HybridSolver::ComputeScaledHybridSystem(const mfem::HypreParMatrix& H)
 {
-    mfem::HypreSmoother prec_scale(const_cast<mfem::HypreParMatrix&>(H));
-
-    mfem::Vector zeros(H.Height());
-    zeros = 1e-8;
-    diagonal_scaling_.SetSize(H.Height());
-    diagonal_scaling_ = 1.0;
-
-    mfem::CGSolver cg_scale(comm_);
-    cg_scale.SetMaxIter(rescale_iter_);
-    cg_scale.SetPreconditioner(prec_scale);
-    cg_scale.SetOperator(H);
-    cg_scale.Mult(zeros, diagonal_scaling_);
-
-    auto Scale = VectorToMatrix(diagonal_scaling_);
-    mfem::HypreParMatrix pScale(comm_, H.N(), H.GetColStarts(), &Scale);
-    H_.reset(smoothg::Mult(pScale, H, pScale));
+    if (rescale_iter_ > 0)
+    {
+        mfem::HypreSmoother prec_scale(const_cast<mfem::HypreParMatrix&>(H));
+
+        mfem::Vector zeros(H.Height());
+        zeros = 1e-8;
```

Comment:
This seems like a strange choice for right-hand side. I guess the idea is to approximate a "near nullspace", so you want to solve something close to `Ax = 0`? But this is actually equivalent to solving `Ax = 1` I think. (I realize this choice does not belong to this PR, so it doesn't change my review.)

Reply:
Yes, I agree that the current setup is equivalent to solving `Ax = 1` (if solve exactly). I now change the rhs to exact 0.